### PR TITLE
feat: holesky deploy scripts

### PR DESCRIPTION
**Pull request by 8sunyuan**
Holesky deployment scripts. Assumes deployment of permissioned multisigs and beaconChainOracle contract. Deployment config is in `script/configs/holesky/M2_deploy_from_scratch.holesky.config.json`. Some config constants for holesky found here https://github.com/eth-clients/holesky

Run script with command
```
forge script script/deploy/holesky/M2_Deploy_From_Scratch.s.sol --rpc-url $RPC_HOLESKY --private-key $PRIVATE_KEY --broadcast -vvvv
```

Locally with anvil node 
```
anvil --fork-url $RPC_HOLESKY
forge script script/deploy/holesky/M2_Deploy_From_Scratch.s.sol --rpc-url http://127.0.0.1:8545 --private-key $PRIVATE_KEY --broadcast -vvvv
```

### CODE REVIEW

File: script/configs/holesky/M2_deploy_from_scratch.holesky.config.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
+{
+  "chainInfo": {
+    "chainId": 17000
+  },
+  "multisig_addresses": {
+    "pauserMultisig": "0x53410249ec7d3a3F9F1ba3912D50D6A3Df6d10A7",
+    "communityMultisig": "0xCb8d2f9e55Bc7B1FA9d089f9aC80C583D2BDD5F7",
+    "operationsMultisig": "0xfaEF7338b7490b9E272d80A1a39f4657cAf2b97d",
+    "executorMultisig": "0x28Ade60640fdBDb2609D8d8734D1b5cBeFc0C348",
+    "timelock": "0xcF19CE0561052a7A7Ff21156730285997B350A7D"
```
Comment (ChaoticWalrus) [inline script/configs/holesky/M2_deploy_from_scratch.holesky.config.json:10]: looks like this contract isn't verified in the explorer?
https://holesky.etherscan.io/address/0xcF19CE0561052a7A7Ff21156730285997B350A7D#code
just noticing this; would be good to take care of so it's not forgotten

File: script/configs/holesky/M2_deploy_from_scratch.holesky.config.json [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      },
+      {
+        "token_address": "0x7322c24752f79c05FFD1E2a6FCB97020C1C264F1",
+        "token_name": "Rocket Pool ETH",
+        "token_symbol": "rETH"
+      }
+    ]
+  },
+  "strategyManager": {
+    "init_strategy_whitelister": "",
```
Comment (ChaoticWalrus) [inline script/configs/holesky/M2_deploy_from_scratch.holesky.config.json:31]: make this the ops multisig, I think?

File: script/configs/holesky/M2_deploy_from_scratch.holesky.config.json [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  },
+  "avsDirectory": {
+    "init_paused_status": 0
+  },
+  "slasher": {
+    "init_paused_status": 0
+  },
+  "eigenPod": {
+    "MAX_RESTAKED_BALANCE_GWEI_PER_VALIDATOR": 32000000000000000000,
+    "GENESIS_TIME": 1695902400
```
Comment (ChaoticWalrus) [inline script/configs/holesky/M2_deploy_from_scratch.holesky.config.json:46]: confirmed this is the correct time 👍

File: script/configs/holesky/M2_deploy_from_scratch.holesky.config.json [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  },
+  "eigenPodManager": {
+    "init_paused_status": 0,
+    "deneb_fork_timestamp": "1707323664"
```
Comment (ChaoticWalrus) [inline script/configs/holesky/M2_deploy_from_scratch.holesky.config.json:50]: where's this value from?
https://github.com/eth-clients/holesky
has `1707305664` -- I guess it's not that important either way though

Comment (8sunyuan) [reply script/configs/holesky/M2_deploy_from_scratch.holesky.config.json:50]: Ahh got it from here https://blog.ethereum.org/2024/01/24/sepolia-holesky-dencun
Will update it regardless.

File: script/configs/holesky/M2_deploy_from_scratch.holesky.config.json [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        "token_address": "0x3F1c547b21f65e10480dE3ad8E19fAAC46C95034",
+        "token_name": "Liquid staked Ether 2.0",
+        "token_symbol": "stETH"
+      },
+      {
+        "token_address": "0x7322c24752f79c05FFD1E2a6FCB97020C1C264F1",
+        "token_name": "Rocket Pool ETH",
+        "token_symbol": "rETH"
+      }
+    ]
```
[the remark this answers is not in the document]
Comment (8sunyuan) [reply script/configs/holesky/M2_deploy_from_scratch.holesky.config.json:28]: At least for our current mainnet strategies, these were the only ones I could find on Holesky

File: script/configs/holesky/M2_deploy_from_scratch.holesky.config.json (hunk 2 above)
Comment (8sunyuan) [reply script/configs/holesky/M2_deploy_from_scratch.holesky.config.json]: https://github.com/Layr-Labs/eigenlayer-contracts/pull/473/commits/4f53c1cdb7159ce3c19cd7a4f113d596dbd2882f

File: script/configs/holesky/M2_deploy_from_scratch.holesky.config.json (hunk 1 above)
Comment (8sunyuan) [reply script/configs/holesky/M2_deploy_from_scratch.holesky.config.json:10]: https://holesky.etherscan.io/address/0xcF19CE0561052a7A7Ff21156730285997B350A7D#code
verified now

File: script/deploy/holesky/M2_Deploy_From_Scratch.s.sol [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+            )
+        );
+        // StrategyManager
+        eigenLayerProxyAdmin.upgradeAndCall(
+            TransparentUpgradeableProxy(payable(address(strategyManager))),
+            address(strategyManagerImplementation),
+            abi.encodeWithSelector(
+                StrategyManager.initialize.selector,
+                executorMultisig, //initialOwner
+                executorMultisig, //initial whitelister
```
Comment (ChaoticWalrus) [inline script/deploy/holesky/M2_Deploy_From_Scratch.s.sol:132]: I think I mentioned this above, but I think this should be the 'ops' multisig (at least to match Goerli/mainnet configs)?

File: script/configs/holesky/M2_deploy_from_scratch.holesky.config.json (hunk 4 above)
Comment (ChaoticWalrus) [reply script/configs/holesky/M2_deploy_from_scratch.holesky.config.json]: OK, LGTM now, thanks 👍

File: script/deploy/holesky/M2_Deploy_From_Scratch.s.sol [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+        _verifyInitializationParams();
+
+        logAndOutputContractAddresses("script/output/holesky/M2_deploy_from_scratch.holesky.config.json");
+    }
+
+    /**
+     * @notice Deploy EigenLayer contracts from scratch for Holesky
+     */
+    function _deployFromScratch() internal {
+        // Deploy ProxyAdmin, later set admins for all proxies to be executorMultisig
```
Comment (ChaoticWalrus) [inline script/deploy/holesky/M2_Deploy_From_Scratch.s.sol:39]: @stevennevins should this be merged into some other script?

**Review by ChaoticWalrus — APPROVED** [review]
This is looking good to me; thank you for putting it together

File: script/deploy/holesky/M2_Deploy_From_Scratch.s.sol (hunk 6 above)
Comment (8sunyuan) [reply script/deploy/holesky/M2_Deploy_From_Scratch.s.sol:132]: ahh missed this, good catch

Comment (8sunyuan) [reply script/deploy/holesky/M2_Deploy_From_Scratch.s.sol]: https://github.com/Layr-Labs/eigenlayer-contracts/pull/473/files#diff-dbf2ec4102ab841c3a08fabf952cdfe8d391f2b48cef41f0e29e5e6740be800aR132

**Pull request merged**